src/components/HomeScreen.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import FlightList from "./FlightList";
+import useFlightApi from "../Apis";
+
+const HomeScreen = () => {
+  const history = useNavigate();
+
+  const [flightList, setFlightList] = useState([]);
+  const [filteredFlights, setFilteredFlights] = useState([]);
+  const [errorApi, setError] = useState(null); // State to store error information
+  const [showLoading, setShowLoading] = useState(true);
+
+  const { data, loading, error } = useFlightApi(
+    "https://flight-status-mock.core.travelopia.cloud/flights"
+  );
+
+  useEffect(() => {
+    if (!loading) {
```

Comment:
again. don't copy state variables that can be used as is. loading, flightlist, error can be used from the hook response itself.

Reply:
have to use state here since we are making api call after every 30 sec, and can;t call it  from inside useEffect.